File: client/database/src/main/java/hr/foi/air/database/entities/Message.java
package hr.foi.air.database.entities;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;
import androidx.room.TypeConverters;

import java.util.Date;

import hr.foi.air.database.converters.DateConverter;

@Entity(tableName = "messages")
@TypeConverters(DateConverter.class)
public class Message {
    @PrimaryKey(autoGenerate = true)
    int id;
    @ForeignKey(entity = Chat.class, parentColumns = "id", childColumns = "chat_id")
    @ColumnInfo(index = true, name = "chat_id")
    String chatId;
    @ForeignKey(entity = User.class, parentColumns = "id", childColumns = "sender_id")
    @ColumnInfo(index = true, name = "sender_id")
    int SenderId;
    @ColumnInfo(name = "send_time")
    Date sendTime;

    String text;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getChatId() {
        return chatId;
    }

    public void setChatId(String chatId) {
        this.chatId = chatId;
    }

    public int getSenderId() {
        return SenderId;
    }

    public void setSenderId(int senderId) {
        SenderId = senderId;
    }

    public Date getSendTime() {
        return sendTime;
    }

    public void setSendTime(Date sendTime) {
        this.sendTime = sendTime;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }
}

File: server/database/entities/Message.php
<?php

class Message {
    private $id;
    private $sendTime;
    private $text;
    private $chatId;
    private $senderId;
    
    public function getId() {
        return $this->id;
    }

    public function getSendTime() {
        return $this->sendTime;
    }

    public function getText() {
        return $this->text;
    }

    public function getChatId() {
        return $this->chatId;
    }

    public function getSenderId() {
        return $this->senderId;
    }

    public function setId($id): void {
        $this->id = $id;
    }

    public function setSendTime($sendTime): void {
        $this->sendTime = $sendTime;
    }

    public function setText($text): void {
        $this->text = $text;
    }

    public function setChatId($chatId): void {
        $this->chatId = $chatId;
    }

    public function setSenderId($senderId): void {
        $this->senderId = $senderId;
    }
    
    public static function fromObject($obj) {
        $message = new Message();
        if (isset($obj->id)) {
            $message->id = $obj->id;
        }
        if (isset($obj->sendTime)) {
            $message->sendTime = $obj->sendTime;
        }
        if (isset($obj->text)) {
            $message->text = $obj->text;
        }
        if (isset($obj->chatId)) {
            $message->chatId = $obj->chatId;
        }
        if (isset($obj->senderId)) {
            $message->senderId = $obj->senderId;
        }
        return $message;
    }

}